homeassistant/components/climate/trigger.py

```diff
@@ -1,16 +1,47 @@
 """Provides triggers for climates."""
 
+import voluptuous as vol
+
+from homeassistant.const import CONF_OPTIONS
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
 from homeassistant.helpers.trigger import (
+    ENTITY_STATE_TRIGGER_SCHEMA_FIRST_LAST,
+    EntityTargetStateTriggerBase,
     Trigger,
+    TriggerConfig,
     make_entity_target_state_attribute_trigger,
     make_entity_target_state_trigger,
     make_entity_transition_trigger,
 )
 
 from .const import ATTR_HVAC_ACTION, DOMAIN, HVACAction, HVACMode
 
+CONF_HVAC_MODE = "hvac_mode"
+
+HVAC_MODE_CHANGED_TRIGGER_SCHEMA = ENTITY_STATE_TRIGGER_SCHEMA_FIRST_LAST.extend(
+    {
+        vol.Required(CONF_OPTIONS): {
+            vol.Required(CONF_HVAC_MODE): vol.All(cv.ensure_list, [HVACMode]),
```

Comment:
Should we require that the list has at least one item?

Reply:
Yes, that's exactly what will happen. OK, I'll change so it requires at least one list item 👍